Nos/Views/GoldenPostView.swift

```diff
@@ -68,23 +69,36 @@ struct GoldenPostView: View {
             let parsedAttributedContent = await Event.attributedContent(
                 noteID: note.identifier,
                 context: backgroundContext
-            ) 
+            )
             withAnimation(.easeIn(duration: 0.1)) {
                 self.noteContent = .loaded(parsedAttributedContent)
             }
         }
     }
-    
+
     var isTextOnly: Bool {
-        (try? note.content?.findUnformattedLinks().count ?? 0) == 0
+        if let content = note.content {
+            return (try? urlParser.findUnformattedURLs(in: content).count) == 0
+        } else {
+            return true
+        }
     }
-    
+
+    var firstImageURL: URL? {
+        let content = note.content
```

Comment:
This is a minor style issue, but for future PRs I'd prefer this:

```suggestion
        guard let content = note.content else { return nil }
```

This would eliminate the need for the `if...else` below and I think it's a little easier to read.

Reply:
Nice catch. I also prefer that. I think it was a consequence of moving that code from the `body` without realizing I could rewrite it.